src/ja/common/proxy/command_handler.py

```diff
@@ -17,7 +22,48 @@ class CommandHandler(ABC):
     success property of the Response is True if both validations were passed
     and the Command was executed without error, and is false otherwise.
     """
-    def __init__(self, socket: str):
+    def __init__(self, socket_path: str):
         """!
-        @param socket: The Unix named socket to listen for Commands on.
+        @param socket_path: The Unix named socket to listen for Commands on.
         """
+        self._socket_path = socket_path
+        self._listen_thread = Thread(target=self._listen)
```

Comment:
I thought the main loop of the server/worker is going to be the command handler, so no need for additional threads here?

Reply:
I think you may be right, but I think a design like this works better.
We're going to have multiple components that need their own thread so I think we should not put the main thread into any of them.

Reply:
so what is the main thread going to do then?

Reply:
Wait, what did we agree on for starting/stopping the server again?
Didn't we need the main thread for that?

Reply:
I am not sure we said anything about starting/stopping it.
How is it going to be done?

Reply:
Or maybe we could discuss this in the meeting today.

Reply:
I think we said that we would make it a systemd service.
We would not need the main thread to control the shutdown in that case, right?

Reply:
So we can just do a while True: sleep(1000) in the main loop .. fine by me.